### PR TITLE
Set R_GC_MEM_GROW=3 globally for all tasks.

### PR DESCRIPTION
**Pull request by plietar**
We've found that setting this value provides significant speedup for some of our workloads. The downside is a slight increase in memory consumption, but our jobs are not usually memory bound anyway.

This can still be overriden, either by setting the `hipercow.default_envvars` option, or by assigning a specific value on individual tasks.

### CODE REVIEW

**Comment by plietar** [comment]
Here are some rough benchmarks on malariasimulation, for a population size of 1M and 1000 time steps: [plot.pdf](https://github.com/mrc-ide/hipercow/files/15419749/plot.pdf)

It's not really clear why `R_GC_MEM_GROW=0` is also beneficial, but it feels to me like a dangerous road to pick.

File: R/envvars.R [hunk 1]
```diff
@@ -69,6 +69,9 @@ c.hipercow_envvars <- function(...) {
     cli::cli_abort("Can't combine 'hipercow_envvars' objects and other objects")
   }
   ret <- rlang::inject(rbind(!!!inputs))
+  ret <- ret[!duplicated(ret$name, fromLast=TRUE),]
```
Comment (richfitz) [inline R/envvars.R:72]:
```suggestion
  ret <- ret[!duplicated(ret$name, fromLast = TRUE), ]
```

File: tests/testthat/test-envvars.R [hunk 2]
```diff
@@ -67,6 +67,15 @@ test_that("can concatenate environment variables", {
 })
 
 
+test_that("can override environment variables when concatenating", {
+  e1 <- hipercow_envvars(A = "1", B = "x")
+  e2 <- hipercow_envvars(A = "2")
+  e3 <- hipercow_envvars(A = "2", secret=TRUE)
```
Comment (richfitz) [inline tests/testthat/test-envvars.R:73]:
```suggestion
  e3 <- hipercow_envvars(A = "2", secret = TRUE)
```

File: R/zzz.R [hunk 3]
```diff
@@ -1 +1,11 @@
...
+
+##' The default set of environment variables used for all tasks.
+##'
+##' These are chosen to provide the best general experience. The default
+##' variables may be overriden globally by setting the hipercow.default_envvars
+##' option, or on a per-task and variable basis by assigning a different value
+##' to the environment variable in question.
+##'
+##' @keywords internal
+DEFAULT_ENVVARS <- hipercow_envvars(R_GC_MEM_GROW = "3")
```
Comment (richfitz) [inline R/zzz.R:11]:
```suggestion
DEFAULT_ENVVARS <- hipercow_envvars("R_GC_MEM_GROW" = "3")
```

**Review by richfitz — CHANGES_REQUESTED** [review]
Can you add a section to the details vignette on this please? https://mrc-ide.github.io/hipercow/articles/details.html

File: R/zzz.R [hunk 4]
```diff
@@ -1 +1,11 @@
...
+
+##' The default set of environment variables used for all tasks.
+##'
+##' These are chosen to provide the best general experience. The default
+##' variables may be overriden globally by setting the hipercow.default_envvars
+##' option, or on a per-task and variable basis by assigning a different value
+##' to the environment variable in question.
+##'
+##' @keywords internal
+DEFAULT_ENVVARS <- hipercow_envvars("R_GC_MEM_GROW" = "3")
```
Comment (richfitz) [inline R/zzz.R:11]:
```suggestion
DEFAULT_ENVVARS <- hipercow_envvars("R_GC_MEM_GROW" = "3") # nolint
```

**Comment by plietar** [comment]
> Can you add a section to the details vignette on this please? https://mrc-ide.github.io/hipercow/articles/details.html

Done.

**Review by richfitz — APPROVED** [review]

**Pull request merged**